wazuh/wazuh
# Add an iam role session duration parameter to AWS module

|Related issue|
|---|
|Closes #9086 |
## Description
In this PR we have added a new parameter to the AWS integration module that allows users to set the duration of the session assumed when using the `--iam_role_arn` parameter. After the changes, the user can specify a session duration between 900 and 3600 seconds using either the `--duration-seconds` parameter or modifyng the `ossec.conf` file.



## Tests
### Manual tests performed
To test the changes we temporary added some code to the module that would add a delay after the `assume_role` call to test if the module would return a timeout when trying to use a session after it has expired.
The execution did return an error, as expected.
### Testing a bucket
```
root@ff5389d2ebf2:/var/ossec/wodles/aws# /var/ossec/framework/python/bin/python3 ./aws-s3.py -b cloudtrail-bucket -t cloudtrail -ds 900 -i arn:aws:iam::xxxxxxx:role/test_role
Time now: 09:03:43
Role assumed. role_session_duration variable is set to 900. Sleeping for 910 seconds.
Time after sleep: 09:18:53
ERROR: Invalid credentials to access S3 Bucket
```
### Testing a service
```
root@ff5389d2ebf2:/var/ossec/wodles/aws# py ./aws-s3.py -sr inspector -ds 900 -i arn:aws:iam::xxxxxxx:role/test_role
Time now: 10:37:26
Role assumed. role_session_duration variable is set to 900. Sleeping for 910 seconds.
Time after sleep: 10:52:36
ERROR: An error occurred (ExpiredTokenException) when calling the ListFindings operation: The security token included in the request is expired
```

We have also tested that the new parameter works fine in the configuration of both a service and a bucket by adding the following configuration items (one at a time) to the `ossec.conf` file and modifying the python program to write the value of the `role_session_variable`, if specified, to a temporary file. 
#### Bucket testing configuration
```
<wodle name="aws-s3">
  <disabled>no</disabled>
  <interval>10m</interval>
  <run_on_start>yes</run_on_start>
  <skip_on_error>yes</skip_on_error>
  <bucket type="cloudtrail">
     <name>wazuh-test-bucket</name>
     <access_key>XXXXXX</access_key>
     <secret_key>XXXXXX</secret_key>
     <iam_role_arn>arn:aws:iam::XXXXX:role/test_role</iam_role_arn>
     <role_session_duration>1300</role_session_duration>
  </bucket>  
</wodle>
```
#### Service testing configuration
```
<wodle name="aws-s3">
  <disabled>no</disabled>
  <interval>1m</interval>
  <run_on_start>yes</run_on_start>
  <skip_on_error>yes</skip_on_error>
  <service type="cloudwatchlogs">
     <access_key>XXXXXX</access_key>
     <secret_key>XXXXXX</secret_key>
     <iam_role_arn>arn:aws:iam::XXXXX:role/test_role</iam_role_arn>
     <role_session_duration>1300</role_session_duration>
     <aws_log_groups>framework-test-apache</aws_log_groups>
  </service>  
</wodle>
